File: CPP/film.cpp
#include "Media.cpp"

class Film : public Media {
protected:
    string genre;
    int durasiMenit;
    string rumahProduksi;

public:
    Film() : Media(), genre(""), durasiMenit(0), rumahProduksi("") {}
    
    Film(int id, string judul, int tahun, string hakCipta, string genre, int durasiMenit, string rumahProduksi)
        : Media(id, judul, tahun, hakCipta) {
        this->genre = genre;
        setDurasiMenit(durasiMenit);
        this->rumahProduksi = rumahProduksi;
    }

    string getGenre() const { return genre; }
    void setGenre(const string& genre) { this->genre = genre; }

    int getDurasiMenit() const { return durasiMenit; }
    void setDurasiMenit(int durasiMenit) { this->durasiMenit = max(durasiMenit, 0); }

    string getRumahProduksi() const { return rumahProduksi; }
    void setRumahProduksi(const string& rumahProduksi) { this->rumahProduksi = rumahProduksi; }
};

File: PHP/film.php
<?php

require_once 'Media.php';

class Film extends Media {
    protected string $genre;
    protected int $durasiMenit;
    protected string $rumahProduksi;

    public function __construct(
        int $id = 0, 
        string $judul = "", 
        int $tahun = 0, 
        string $hakCipta = "", 
        string $genre = "", 
        int $durasiMenit = 0, 
        string $rumahProduksi = ""
    ) {
        parent::__construct($id, $judul, $tahun, $hakCipta);
        $this->genre = $genre;
        $this->setDurasiMenit($durasiMenit);
        $this->rumahProduksi = $rumahProduksi;
    }

    public function getGenre(): string { return $this->genre; }
    public function setGenre(string $genre): void { $this->genre = $genre; }

    public function getDurasiMenit(): int { return $this->durasiMenit; }
    public function setDurasiMenit(int $durasiMenit): void { $this->durasiMenit = max($durasiMenit, 0); }

    public function getRumahProduksi(): string { return $this->rumahProduksi; }
    public function setRumahProduksi(string $rumahProduksi): void { $this->rumahProduksi = $rumahProduksi; }
}